Encryption.md

```diff
@@ -19,216 +19,544 @@
+byte array can be specified in a file - to help a reader retrieve the AAD 
+prefix string used for file enciphering. The aad_metadata  field can be empty, since readers might 
+know apriori a right AAD_Prefix for each file. Alternatively, the aad_metadata  can contain hints 
+that help readers to get the right AAD prefix. Yet another option is to keep the AAD_Prefix itself 
+inside the aad_metadata field – in that case, the readers must be able to verify it independently, 
+since this field is not protected in a storage, and can be changed or filled by an attacker.
+
+Here go a number of examples of AAD_Prefix creation and management in an organization:
+
+   * An organization uses a convention is to build an AAD_Prefix from the table name/ID, date and 
+   a partition . For example, a writer of a May 23 version of an employee table will use 
+   "employees_23_05_2018.partN" as an AAD prefix (and as the file name) for each table file. 
+   The reader that needs to process the employee table, a May 23 version, knows the AAD must be 
+   "employees_23_05_2018.partN" 
+   in each corresponding table file. This approach doesn’t use the aad_metadata field.
+   * A similar convention, but the date format (and other details) can be variable. Then the 
+   format is kept in the aad_metadata field, for example as "MM_DD_YY.partitionN". The reader will 
+   know the AAD prefixes are "employees_05_23_18.partition0" etc.
+   * The aad_metadata field contains the AAD prefix itself. In this case, the reader must be able 
+   to verify its validity. For example, if the AAD prefix stored in aad_metadata is "employees_23May2018.part0001", 
+   the reader should know it is fine, but if the aad_metadata stores "employees_23May2016.part0001" - 
+   the version is wrong.
+   * The AAD_Prefix is generated as a random string. Together with a file name, it is sent by the file 
+   writer to a trusted metadata service, deployed in the organization. The reader, that needs to parse a 
+   certain version of a table, authenticates the service and gets the list of table files and their AAD prefixes.
+   * Frameworks above Parquet create table integrity tools that manage the file AAD prefixes. For example, 
+   a framework table writer can create a signature file that contains the table ID, file list/number and 
+   optional AAD prefixes. The AAD prefixes can be either explicit (eg random) or derived from 
+   the table ID and file name/number. The signature file is signed with a secret key (private or symmetric), and 
+   stored next to the table files, or in another location known to the readers. The 
+   readers get the public or symmetric key, verify the signature file and get the AAD prefixes for the table files 
+   (potentially, along with file names/paths).
+   
+#### 4.4.2 AAD_Suffix
+File integrity can be disrupted by module swapping inside the file - such as replacement 
```

Comment:
These sentences are written in a misleading way. They state that "file integrity can be disrupted by module swapping inside the file" first, and only later introduce the AAD_suffix to prevent it.

Instead, this should state that the purpose of the AAD_suffix is to prevent the following integrity attacks, then have a list of some of the swapping attacks. That way, the reader knows that this specification prevents those attacks.

Consider a similar change to the AAD_prefix section above.